Formula/zsh.rb

```diff
@@ -24,11 +24,10 @@ class Zsh < Formula
     depends_on "autoconf" => :build
   end
 
+  depends_on "texinfo" => :build if OS.linux? || MacOS.version >= :ventura
```

Comment:
I think the current way is:
```suggestion
  on_system :linux, macos: :ventura_or_newer do
    depends_on "texinfo" => :build
  end
```

But don't know if we want to discuss new DSL to handle this before propagating syntax to all `texinfo` and `groff` usage.